data_processing/cleaning_demographic_data.Rmd

```diff
@@ -52,11 +50,412 @@ demo_dat %>%
+                                      str_starts(race_1, "wh") ~ "white",
+                                   str_detect(race_1, "unkn|#|client|collected") ~ "unknown",
+                                   is.na(race_1) ~ "unknown",
+                                      
+                                         TRUE ~ race_1))
+```
+
+check
+```{r}
+demo_dat %>% 
+  count(race_1, race_1_recode)
+```
+
+race_2 - doing the same as for race_1, and replacing no applicable with not applicable, updating "unknown". For those who put an ethnicity response in race_2, replacing with "unknown" and updating those in the ethnicity variable. 
+
+```{r}
+demo_dat <- demo_dat %>% 
+  mutate(race_2_recode = case_when(str_starts(race_2,"americ") ~ "american indian/alaskan native/indigenous", 
+                                      str_starts(race_2, "asian") ~ "asian/asian american",
+                                      str_starts(race_2, "bla") ~ "black/african american/african",
+                                      str_starts(race_2, "pacific i") ~ "native hawaiian/pacific islander",
+                                      str_starts(race_2, "wh") ~ "white",
+                                   str_detect(race_2, "hisp") ~ "unknown",
+                                   str_detect(race_2, "unknown|unable|refused|not collected|applicable|verified|#|client|mex") ~ "unknown",
+                                   is.na(race_2) ~ "unknown",
+                                      
+                                         TRUE ~ race_2))
+```
+
+check for race_2
+```{r}
+demo_dat %>% 
+  count(race_2, race_2_recode)
+```
+
+
+Ethnicity - replace all instances of different spellings or orders of hispanic/latino/a/x to hispanic/latinx. Changing mexican/chicano, cuban, other hispanic/latino, and puerto rican to hispanic/latinx. Also, replace all instances of different spellings or orders of not hispanic/latino to non-hispanic/latinx 
+
+```{r}
+demo_dat <- demo_dat %>% 
+  mutate(ethnicity_recode = case_when(str_starts(ethnicity, "no") ~ "non-hispanic/latinx",
+                                      str_detect(ethnicity,"hispa") ~ "hispanic/latinx", 
+                                      str_detect(race_2, "not hispanic|no hispanic") ~ "non-hispanic/latinx",
+                                      str_detect(race_2, "hispanic") ~ "hispanic/latinx",
+                                      str_detect(race_2, "mex") ~ "hispanic/latinx",
+                                      str_detect(ethnicity, "mex|puerto|cuban") ~ "hispanic/latinx",
+                                      str_detect(ethnicity, "unknown|#|applicable|data not|client|missing|rent") ~ "unknown",
+                                      is.na(ethnicity) ~ "unknown",
+                                      
+                                      
+                                         TRUE ~ ethnicity))
+```
+
+check for ethnicity 
+```{r}
+demo_dat %>% 
+  count(ethnicity, race_2, ethnicity_recode) 
+```
+
```

Comment:
Great sanity check--the code above to recode looks good and this shows that they fell in the right buckets!